libc-top-half/musl/src/thread/pthread_self.c

```diff
@@ -1,6 +1,12 @@
 #include "pthread_impl.h"
 #include <threads.h>
 
+#if defined(_REENTRANT)
```

Comment:
`if defined(__wasm__) &&  defined`_REENTRANT)`

Also wrap in `__wasilibc_unmodified_upstream`?